src/CloudSeis.jl

```diff
@@ -478,6 +512,31 @@ function make_extents(containers::Vector{C}, nextents, foldername, nominal_frame
     extents
 end
 
+function validate_history_dictionary(h)
+    for key in keys(h)
+        if key ∉ ("inputs", "flow")
+            error("history dictionary has invalid key '$key'")
+        end
+    end
+
+    for key in keys(get(h, "flow", Dict()))
+        if key ∉ ("processes", "parameters")
+            error("history[\"flow\"] dictionary has invalid key '$key'")
+        end
+    end
+
+    processes = get(get(h, "flow", Dict()), "processes", [])
+    for (iprocess,process) in enumerate(processes)
+        for key in keys(process)
+            if key ∉ ("process", "parameters")
+                error("history[\"flow\"][\"processes\"][$iprocess] has invalid key '$key'")
+            end
+        end
+    end
+
```

Comment:
Check if each element of history[inputs] has invalid key (a key other than "container" and "history")?